browser/tor/sources.gni

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import("//brave/components/tor/buildflags/buildflags.gni")
+
+brave_browser_tor_sources = []
+brave_browser_tor_deps = []
+
+if (enable_tor) {
+  brave_browser_tor_sources = [
+    "//brave/browser/tor/tor_profile_manager.cc",
+    "//brave/browser/tor/tor_profile_manager.h",
+    "//brave/browser/tor/tor_profile_service_factory.cc",
+    "//brave/browser/tor/tor_profile_service_factory.h",
+  ]
+
+  brave_browser_tor_deps = [
+    "//base",
+    "//brave/components/brave_webtorrent/browser/buildflags",
+    "//brave/components/constants",
+    "//brave/components/tor",
+    "//brave/components/tor:common",
+    "//brave/components/tor:pref_names",
+    "//chrome/browser/profiles",
+    "//chrome/common",
+    "//components/autofill/content/browser",
+    "//components/autofill/core/browser",
+    "//components/keyed_service/content",
+    "//components/safe_browsing/core/common:safe_browsing_prefs",
+    "//components/translate/core/browser",
+    "//components/user_prefs",
+    "//content/public/browser",
+    "//net",
+    "//third_party/blink/public/common",
+    "//url",
+  ]
+
+  # Below dep list are not directly used tor target.
+  # Only added to fix intermittent build failure.
+  # Most of the case comes from including brave_browser_process_impl.h`
+  # headers. That header includes some buildflags but sometimes they are
+  # not generated before evaluating tor target.
+  brave_browser_tor_deps += [
```

Comment:
@bridiver I think should be safe to drop these now when the code is in source.gni ?

Reply:
also  the include for brave_browser_process_impl.h in brave_vpn_button_unittest.cc is wrong and should be replaced with `brave_browser_process.h` and depend on `//brave/browesr:browser_process`

Reply:
brave_browser_process_impl.h seem to be unused in the file. Removed.